src/components/outline-nav/components/outline-nav-with-active/index.tsx

```diff
@@ -0,0 +1,65 @@
+import { useMemo } from 'react'
+import OutlineNav from 'components/outline-nav'
+import { OutlineLinkItem } from 'components/outline-nav/types'
+import {
+	getItemSlugs,
+	highlightActiveItems,
+} from 'components/outline-nav/utils'
+import { useDeviceSize } from 'contexts'
+import { useActiveSection } from 'lib/hash-links/use-active-section'
+
+/**
+ * Given an array of OutlineLinkItems,
+ * render an outline nav, and automatically highlight the active item.
+ */
+function OutlineNavWithActive({
+	items,
+}: {
+	items: Omit<OutlineLinkItem, 'isActive'>[]
+}) {
+	/**
+	 * We useDeviceSize to avoid running active heading calculations
+	 * if we're on a viewport where we won't render the nav anyways.
+	 *
+	 * TODO: isDesktop isn't fully accurate, we may want to adjust this
+	 * based on the specific breakpoint at which we show the sidecar.
+	 *
+	 * Thought: could we monitor the visibility of `OutlineNav` instead,
+	 * ie the root element we're rendering in this component?
+	 * If it's visible in CSS, then we should enable useActiveSection.
+	 * If it's not visible, then we should NOT enable useActiveSection.
+	 * Maybe a `useVisibility` hook or something would be useful here.
+	 * Maybe this could reduce reliance on specific breakpoint metrics.
```

Comment:
This could be cool if we want to keep this component aware of the layout level scaling. Personally I think it would be ideal for the layout to decide whether this is `enabled` based on whatever logic applies within that context and pass that down? Then the outline isn't concerned with layout level breakpoints. Just an idea, and maybe doesn't make as much sense in practice when we actually use this within the layout slot. 

 Also, if this is to prevent extra calculations for the active state, depending on how we _hide_ the sidecar at smaller breakpoints, would that (assuming its `display:none`) take care of this issue?

Reply:
Yeah good call - I've added a follow-up task 🎟 [Rethink SidebarSidecarLayout responsive implementation to avoid useActiveSection calculations, at layout level](https://app.asana.com/0/0/1204024966433307/f) to address this. It seems like more of an optimization, so thinking I'll dive into it only if I have time, but open to other approaches 🙇

Re: hiding with `display: none`, I think the hard part is we can't call hooks conditionally. So even if the CSS says hide the sidecar, React will still be rendering it (with that CSS) and still make the `useActiveSection` calls. So +1 to having the layout decide when `useActiveSection` should be running or not 👍